### PR TITLE
Feature/failed hooks report

### PR DESCRIPTION
**Pull request by ramiloif**
Resolve for #150 issue .
Added support for all hook failures :
- before all
- before each
- after all 
- after each

The hook will be shown like a test failure but will not affect the global statistic. 
Hook will be shown only when it failed.

The result after integrating with my PR on https://github.com/adamgruber/mochawesome-report-generator/pull/28
screen shot 2017-04-30 at 0 17 54
Full html report is in here :
https://gist.github.com/ramiloif/b0846b4aca19ce1a50a12260151c10b4

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -80,4 +80,4 @@
     "should": "^11.1.2",
     "sinon": "^2.1.0"
   }
-}
+}
```
Comment (adamgruber) [inline package.json]: Nit: keep newline at end of file

File: test/sample-suite.js [hunk 2]
```diff
@@ -22,7 +22,6 @@ module.exports = {
           _retries: -1,
           _currentRetry: 0,
           pending: false,
-          type: 'test',
```
Comment (adamgruber) [inline test/sample-suite.js]: Why was `type` removed? It is part of the raw test object so should be here. If you need to, add new objects for hooks with `type: 'hook'`.

**Comment by ramiloif** [comment]
Hi i i fixed all nits ,
waiting for review.

**Comment by ramiloif** [comment]
Why dist/ isn't in the .gitignore ?

**Comment by adamgruber** [comment]
`dist/` isn't ignored because I haven't set up an automatic build process but I generate it before publishing. Adding it to the PR adds unnecessary noise.

File: test/sample-tests.js [hunk 3]
```diff
@@ -16,7 +16,6 @@ module.exports = {
       _retries: -1,
       _currentRetry: 0,
       pending: false,
-      type: 'test',
```
Comment (adamgruber) [inline test/sample-tests.js]: Still not sure why you're removing the `type` property in these test objects?

Comment (ramiloif) [inline test/sample-tests.js]: Fixed ! 
It's raw object so it should not be changed my bad ....

**Pull request merged**